docs: add usePreviewMode explanation (#26602)
nuxt/nuxt: docs/3.api/2.composables/use-preview-mode.md

# `usePreviewMode`

Preview Mode allows you to see how your changes would be displayed on a live site without revealing them to users.

You can use the built-in `usePreviewMode` composable to access and control preview state in Nuxt. If the composable detects preview mode it will automatically force any updates necessary for [`useAsyncData`](/docs/api/composables/use-async-data) and [`useFetch`](/docs/api/composables/use-fetch) to rerender preview content.

```js
const { enabled, state } = usePreviewMode()
```

## Example

The example below creates a page where part of a content is rendered only in `Preview mode`.

```vue [pages/some-page.vue]
<script setup>
const route = useRoute()
const { enabled, state } = usePreviewMode({
  shouldEnable: () => {
    return route.query.customPreview === 'true'

>   },

Reviewer: maybe it would be worth leaving the example below as an example of `shouldEnable` and modify this one so it shows how it works out of the box, without `shouldEnable`?

Author: I've been thinking about this while making my PR and the idea was to show in the example as much as possible. But I understand that might be overwhelming. I will update the example. Thank you

Author: I think it's not necessary to have whole example at the bottom, because there already is an example  for `shouldEnable` :

```
export function useMyPreviewMode () {
  return usePreviewMode({
    shouldEnable: () => {
      return !!route.query.customPreview
    }
  });
}
```

Reviewer: sounds good

})
const { data } = await useFetch('/api/preview', {
  query: {
    apiKey: state.token
  }
})
</script>
<template>
  <div>
    Some base content
    <p v-if="enabled">
      Only preview content: {{ state.token }}
      <br>
      <button @click="enabled = false">
        disable preview mode
      </button>
    </p>
  </div>
</template>
```

Now you can generate your site and serve it

```bash [Terminal]
npx nuxi generate
npx serve .output/public
```

Then you can see your preview page by adding the query param defined in `shouldEnable` to the end of the page you want to see once:

```js
?customPreview=true
```

::note

> usePreviewMode should be tested locally with nuxi generate and not yarn dev

Reviewer: ```suggestion
`usePreviewMode` should be tested locally with `nuxi generate` and then `nuxi preview` rather than `nuxi dev`.
```

Author: Thank you for your suggestion. I will update the note, but I will use **nuxi start**, because for junior developer, it might seem that **nuxi preview** is somehow related to usePreviewMode and might be surprised why they don't see preview code right away
